components/MainSearchBar.jsx

```
@@ -61,73 +69,128 @@ const MainSearchBar = ({ onLocationSelect, onBuildingPress }) => {
                 interimResults: false,
                 maxAlternatives: 1,
             });
-
-            setTimeout(() => setIsListening(false), 5000);
         } catch (error) {
             setIsListening(false);
+
         }
     };
 
+    const fetchPredictions = async (text) => {
```

Handle edge cases like empty predictions or network failures more gracefully like showing "no results" instead of an empty list

---

I think we can do it using try/catch or validate if there are any predictions before we start working on it!